lib/chef/resource/windows_script.rb

```diff
@@ -31,8 +31,8 @@ class WindowsScript < Chef::Resource::Script
       def initialize(name, run_context, resource_name, interpreter_command)
         super(name, run_context)
         @interpreter = interpreter_command
-        @resource_name = resource_name
-        @default_guard_interpreter = resource_name
+        @resource_name = resource_name if resource_name
```

Comment:
This is a different case because it isn't an optional arg.  This one is about the subclass passing resource_name--and I have changed the subclasses to pass `nil` since resource_name is already being set automatically.  However, I didn't want to remove the argument entirely, because I don't know if someone else somewhere may be overriding this and passing 4 args.